src/Renderer/Camera.js

```diff
@@ -100,4 +101,26 @@ Camera.prototype.box3SizeOnScreen = function box3SizeOnScreen(box3, matrixWorld)
     return tmp.box3;
 };
 
+ /**
+ * Test for collision between camera and a geometry layer (DTM/DSM) to adjust camera position
+ * It could be modified later to handle an array of geometry layers
+ * @param {view} view where we test the collision between geometry layers and the camera
+ * @param {elevationLayer} elevationLayer (DTM/DSM) used to test the collision with the camera. Could be another geometry layer
+ * @param {minDistanceCollision} minDistanceCollision the minimum distance allowed between the camera and the surface
+ */
+Camera.prototype.adjustAltitudeToAvoidCollisionWithLayer = function adjustAltitudeToAvoidCollisionWithLayer(view, elevationLayer, minDistanceCollision) {
+    const camLocation = new Coordinates(view.referenceCrs, view.camera.camera3D.position).as('EPSG:4326');
```

Comment:
`EPSG:4326` is wrong here (are you sure you tried in planar mode :) ?)

But the good news is: there's a ready to use function that does what you need: 
```js
const camLocation = view.camera.position(); 
```

Reply:
No EPSG:4326 is geographic which is needed.

Reply:
(And yes of course I tested in planar and it works)

Reply:
Good to know that it's working but... it doesn't make sense, right?
Why would you transform from a planar CRS to geographic to query elevation in a layer with a planar CRS?

Reply:
Yes it looks weird:) That's because we can easily manipulate the altitude in geographic coord using the coordinates class.
But yes I can simplify it to 
```javascript
const camLocation = view.camera.position().as('EPSG:4326');
```
and better, we could also modify Coordinates class so we handle altitude as a height if Coord is not geographic. This would avoid coordinate transform when in projected coordinate system

Reply:
I'll rebase and add a comment to the function so it's clearer

Reply:
Ok, should be enough for now.